src/modules/layoutUiRules/rulesUtils.js

```diff
@@ -0,0 +1,59 @@
+import { getLockedDimsByVisType } from './rules'
+
+import {
+    getAxisMaxNumberOfDimsByVisType,
+    getAxisMaxNumberOfItemsByVisType,
+    getAllLockedDimIdsByVisType,
+} from './rulesHelper'
+
+export const hasAxisTooManyItemsByVisType = (
+    visType,
+    axisId,
+    numberOfItems
+) => {
+    const maxNumberOfItemsPerAxis = getAxisMaxNumberOfItemsByVisType(
+        visType,
+        axisId
+    )
+
+    return maxNumberOfItemsPerAxis
+        ? numberOfItems > maxNumberOfItemsPerAxis
+        : false
+}
+
+export const isDimensionLocked = (visType, dimensionId) =>
+    getAllLockedDimIdsByVisType(visType).find(id => id === dimensionId)
+
+export const isAxisFull = (visType, axisId, axisDimensionsCount) =>
+    axisDimensionsCount >= getAxisMaxNumberOfDimsByVisType(visType, axisId)
+
+export const canDimensionBeAddedToAxis = (visType, axis, axisId) => {
+    const axisIsFull = isAxisFull(visType, axisId, axis.length)
+    const transferableDimension = getTransferableDimensionPerAxisByVisType(
+        visType,
+        axisId,
+        axis
+    )
+
+    // 1 dimension allowed in axis
+    // 1 dimension is already present and not locked
+    // the dragged one can be added and will cause the old one to be moved to filters
+    // what happens with max limit > 1, axis full and 1 or more locked dimensions?
+    return !(axisIsFull && !transferableDimension)
+}
+
+export const getTransferableDimensionPerAxisByVisType = (
+    visType,
+    axisId,
+    axis
+) => {
+    const lockedDimsByVisType = getLockedDimsByVisType(visType)
```

Comment:
can you replace `const lockedDimsByVisType = getLockedDimsByVisType(visType)` and `Object.keys(lockedDimsByVisType)` with `getAllLockedDimIdsByVisType` from rules helper and skip the`'./rules'` import?

Reply:
I need the `axisId` information in order to do the filtering...